### DISCUSSION TITLE
Suggestions for improving the program

### DISCUSSION THREAD

**Discussion by artem59k** [BearWare/TeamTalk5#1415] [General]
### Description

Hello. In the window of connecting to the server in the qt client, the NVDA screen reader does not pronounce characters when the left and right arrows are pressed in the fields with a combo list. In linux, Orca says everything without problems when we press the space bar. Can this be fixed? Since this is a critical accessibility issue. We can't check the host address and username correctly. There is also an error when, with an incorrect account, the client asks for a username and password, but when entering 
correct data and after clicking ok, an error window appears again. Is it possible to implement a function to statically specify the folder on the computer where the ffiles will be downloaded from the server, in order not to constantly specify the save path. I also suggest adding the ability to multi-save files for administrators and users, so you can download several files from the server faster.
Is there any way to fix the problem: files are uploaded to the server slowly. If there is a solution, please 
tell me. I suggest adding some rights and user roles. For example, only a few users can connect to the channel, in which case the channel operator can not only block users in the channel, but also add to the list of allowed channel users those who can join the room. I also propose to expand the possibility of creating a channel. For example, who can upload files to this channel: only administrators, users and administrators. I also suggest improving 
hidden channel mode. Which will allow you to add subchannels to the hidden channel. And only when the user gets into the hidden channel, the rest of the channels will be shown to him. This way you can create interest clubs on the server. There is also a suggestion about hidden channels: when the hidden channel checkbox is selected, then the following button appears: who can see this channel? And in this window, you can specify to the administrator: which users to show the channel to. I propose to implement the possibility of copying 
and moving files between channels on the server. I really like this program: and these suggestions will help to improve the software. Thank you! You are doing a great job of developing an independent system for communication.

### Application

- [X] qtTeamTalk
- [X] TeamTalkAndroid
- [X] iTeamTalk
- [X] TeamTalkServer

**Comment by amirmahdifard** [comment]
i like moving files between channels.
